crates/re_space_view_dataframe/src/space_view_class.rs

```diff
@@ -75,112 +84,129 @@ impl SpaceViewClass for DataframeSpaceView {
-            .unique()
+            // TODO(#4466): make that optional
             .filter(|comp| !comp.is_indicator_component())
             .collect();
 
+        // Draw the header row.
+        let header_ui = |mut row: egui_extras::TableRow<'_, '_>| {
+            row.col(|ui| {
+                ui.strong("Entity");
+            });
+
+            for comp in &sorted_components {
+                row.col(|ui| {
+                    ui.strong(comp.short_name());
+                });
+            }
+        };
+
+        // Draw a single line of the table. This is called for each _visible_ row, so it's ok to
+        // duplicate some of the querying.
+        let row_ui = |idx: usize, mut row: egui_extras::TableRow<'_, '_>| {
+            let instance = &sorted_instance_paths[idx];
+
+            // TODO(#4466): make it explicit if that instance key is "out
+            // of bounds" (aka cannot be joined to a primary component).
+
+            row.col(|ui| {
+                instance_path_button(ctx, ui, None, instance);
+            });
+
+            for comp in &sorted_components {
+                row.col(|ui| {
+                    // TODO(#4466): make it explicit if that value results
+                    // from a splat joint.
+
+                    if let Some((_, comp_inst)) =
+                        // This is a duplicate of the one above, but this ok since this codes runs
+                        // *only* for visible rows.
+                        get_component_with_instances(
+                            store,
+                            &latest_at_query,
+                            &instance.entity_path,
+                            *comp,
+                        )
+                    {
+                        ctx.component_ui_registry.ui(
+                            ctx,
+                            ui,
+                            UiVerbosity::Small,
+                            &latest_at_query,
+                            &instance.entity_path,
+                            &comp_inst,
+                            &instance.instance_key,
+                        );
+                    } else {
+                        ui.weak("-");
+                    }
+                });
+            }
+        };
+
         egui::ScrollArea::both()
             .auto_shrink([false, false])
             .show(ui, |ui| {
+                ui.style_mut().wrap = Some(false);
+
                 egui::Frame {
                     inner_margin: egui::Margin::same(5.0),
                     ..Default::default()
                 }
                 .show(ui, |ui| {
                     egui_extras::TableBuilder::new(ui)
                         .columns(
-                            Column::auto_with_initial_suggestion(200.0),
-                            all_components.len() + 1,
+                            Column::auto_with_initial_suggestion(200.0).clip(true),
+                            sorted_components.len() + 1,
```

Comment:
very nitpicky style suggestion: the extra column in the _first_ one

```suggestion
                            1 + sorted_components.len(),
```